src/librustc_mir/build/matches/test.rs

```diff
@@ -162,12 +162,19 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
 
             TestKind::SwitchInt { switch_ty, ref options, indices: _ } => {
-                let otherwise = self.cfg.start_new_block();
-                let targets: Vec<_> =
-                    options.iter()
-                           .map(|_| self.cfg.start_new_block())
-                           .chain(Some(otherwise))
-                           .collect();
+                let targets: Vec<_> = match switch_ty.sty {
+                    ty::TyBool if options.len() == 2
```

Comment:
well, this commit alone is not enough, in that it invalidates the `SwitchInt` terminator's invariants. e.g. this comment on `targets` says:

```rust
         /// Possible branch sites. The length of this vector should be
        /// equal to the length of the `values` vector plus 1 -- the
        /// extra item is the block to branch to if none of the values
        /// fit.
        targets: Vec<BasicBlock>,
```

but here you are not generating the otherwise block. It'd be better to change terminator in that case I imagine?